### PR TITLE
Add config support for "on_error" in Elasticsearch tracing

### PR DESCRIPTION
**Pull request by bill-kolokithas**
This mirrors the functionality that is already present in many other instrumenters.

Reasons for wanting this:

One might want to ignore some errors in some expected scenarios such as

`Elasticsearch::Transport::Transport::Errors::NotFound`
or
`Elasticsearch::Transport::Transport::Errors::Conflict`

**What does this PR do?**
Adds `on_error` configuration support for Elasticsearch tracing and forwards it to `Tracing.trace`

**Motivation:**
We have flows where `NotFound` exceptions are expected and we make sure to handle them in the application.
We don't want them to show up as errors in the dashboard though.

**Change log entry**
Yes. Add configuration option `on_error` to Elasticsearch tracing

(Added by @Strech)

**How to test the change?**
Run provided tests with `bundle exec rake test:elasticsearch`

### CODE REVIEW

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -677,6 +677,7 @@ Datadog.configure_onto(client.transport, **options)
 | `service_name` | `DD_TRACE_ELASTICSEARCH_SERVICE_NAME` | `String` | Name of application running the `elasticsearch` instrumentation. May be overridden by `global_default_service_name`. [See _Additional Configuration_ for more details](#additional-configuration) | `elasticsearch` |
 | `peer_service` | `DD_TRACE_ELASTICSEARCH_PEER_SERVICE` | `String` | Name of external service the application connects to                                                                                                                                              | `nil`           |
 | `quantize`     |                                       | `Hash`   | Hash containing options for quantization. May include `:show` with an Array of keys to not quantize (or `:all` to skip quantization), or `:exclude` with Array of keys to exclude entirely.       | `{}`            |
+| `on_error` | | `Proc` | Custom error handler invoked when a request raises an error. Provided `span` and `error` as arguments. Sets error on the span by default. Useful for ignoring transient errors. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |
```
Comment (rtrieu) [inline docs/GettingStarted.md:680]:
```suggestion
| `on_error` | | `Proc` | Custom error handler invoked when a request raises an error. Provides `span` and `error` as arguments. Sets error on the span by default. Useful for ignoring transient errors. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Minor feedback on tense.

File: docs/GettingStarted.md (hunk 1 above)
Comment (bill-kolokithas) [reply docs/GettingStarted.md]: Thanks! fixup: https://github.com/DataDog/dd-trace-rb/pull/4066/commits/1dc216f35ce8b70deeacf60acc67e0c5d4b74e2b

**Review by rtrieu — APPROVED** [review]

**Review by marcotc — APPROVED** [review]

**Review by Strech — APPROVED** [review]
Well done 👏🏼 

I've updated PR body and add **Change log entry** section

**Pull request merged**